tests/integration_test.rs

```diff
@@ -151,9 +156,9 @@ async fn end_to_end_test() {
         ..Config::default()
     };
 
-    let otel_component = Arc::new(Otel::new(config));
-    let otel_component_clone = otel_component.clone();
-    let otel_long_running_task = tokio::spawn(async move { otel_component_clone.run().await });
+    let otel_component = Arc::new(Mutex::new(Otel::new(config)));
```

Comment:
See if we can avoid having a mutex here - the otel component should be immutable and shareable across tasks.

Reply:
Removed the Arc::Mutex as it wasn't needed here. However, I left run() as taking a mutable ref. This change was needed when I added the shutdown channel because recv() takes a mutable ref. If this becomes a problem we can look into different solutions for graceful shutdown.